client/fingerprint/env_aws.go

```diff
@@ -105,23 +101,19 @@ func (f *EnvAWSFingerprint) Fingerprint(request *FingerprintRequest, response *F
 		"placement/availability-zone": false,
 	}
 	for k, unique := range keys {
-		res, err := client.Get(metadataURL + k)
-		if err != nil {
+		resp, err := ec2meta.GetMetadata(k)
+		if awsErr, ok := err.(awserr.RequestFailure); ok {
+			f.logger.Debug("could not read attribute value", "attribute", k, "error", awsErr)
+			continue
+		} else if awsErr, ok := err.(awserr.Error); ok {
 			// if it's a URL error, assume we're not in an AWS environment
 			// TODO: better way to detect AWS? Check xen virtualization?
```

Comment:
is this TODO and check solved by the introduction of your `ec2meta.Available()` check?

Reply:
I don't believe so.  `ec2meta.Available()` does the same check as before, namely check a metadata entry (instance-id or ami-id).

I'm not fully sure what this protect against and how we expect it to behave.  If the earlier check Available metadata lookup succeeded, but looking up another metadata failed with a connection error, it feels to me we ought to retry rather than give up completely?!

Either way, I decided to leave this alone /shrug.

Reply:
I don't believe so.  `ec2meta.Available()` does the same check as before, namely check a metadata entry (instance-id or ami-id).

I'm not fully sure what this protect against and how we expect it to behave.  If the earlier check Available metadata lookup succeeded, but looking up another metadata failed with a connection error, it feels to me we ought to retry rather than give up completely?!

Either way, I decided to leave this alone /shrug.